server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/projection/api/MailQueueProjection.java

```diff
@@ -0,0 +1,29 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.projection.api;
+
+import java.util.stream.Stream;
+
+import org.apache.james.queue.rabbitmq.projection.api.model.MailQueueItem;
+
+public interface MailQueueProjection {
```

Comment:
A projection reflects apart of it underlying model, In this case It is the projection of `ManageableMailQueue`. You mean I should make it extends `ManageableMailQueue`?

Reply:
I mean that a projection being a way to organize data for a given usage (for example, a mail counter is a projection of my mailbox) what matters is the purpose of the projection. BTW, AFAIK, "projection" term is only used for data storage, so it could be a DAO name but probably not a service name.
In this case, MailQueueProjection looks a lot like ManageableMailQueue, I can't find a name because it looks redundant.